package/lib/data/dito_api_interface.dart

```diff
@@ -0,0 +1,285 @@
+      ..ditoId = user.id ?? ""
+      ..email = user.email ?? ""
+      ..name = user.name ?? ""
+      ..birthday = user.birthday ?? ""
+      ..gender = user.gender ?? ""
+      ..address = (api.UserInfo_Address()
+        ..city = user.address?.city ?? ""
+        ..country = user.address?.country ?? ""
+        ..postalCode = user.address?.postalCode ?? ""
+        ..state = user.address?.state ?? ""
+        ..street = user.address?.street ?? "");
+  }
+
+  Future<bool> _apiCall(List<int> serializedRequest) async {
+    _checkConfiguration();
+
+    final headers = {
+      'Content-Type': 'application/proto',
+      'platform_api_key': _apiKey!,
+      'sha1_signature': _secretKey!,
+    };
+
+    final response = await http.post(
+      Uri.parse(url),
+      headers: headers,
+      body: serializedRequest,
+    );
+
+    if (response.statusCode == 200) {
+      final responseProto = api.Response.fromBuffer(response.bodyBytes);
+      for (var responseData in responseProto.response) {
+        if (responseData.hasError()) {
+          final activityId = responseData.id;
+          final error = responseData.error;
+
+          if (kDebugMode) {
+            print('Activity: $activityId');
+            print('Error Code: ${error.code}');
+            print('Error Message: ${error.message}');
+            switch (error.code) {
+              case api.ErrorCode.ERROR_INVALID_REQUEST:
+                throw('Invalid request format.');
+              case api.ErrorCode.ERROR_UNAUTHORIZED:
+                throw('Unauthorized access.');
+              case api.ErrorCode.ERROR_NOT_FOUND:
+                throw('Resource not found.');
+              case api.ErrorCode.ERROR_INTERNAL:
+                throw('Internal server error.');
+              case api.ErrorCode.ERROR_NOT_IMPLEMENTED:
+                throw('Feature not implemented.');
+              default:
+                throw('Unknown error occurred.');
+            }
+          }
+        }
+      }
+
+      return true;
+    }
+
+    throw('Unknown error occurred.');
+  }
+
+  Future<api.Request> createRequest() async {
+    final device = await getDeviceToken();
+    final sdk = await getSDKInfo();
+    final app = await getAppInfo();
+    final user = getUserInfo();
+
+    return api.Request()
+      ..user = user
+      ..device = device
+      ..sdk = sdk
+      ..app = app;
+  }
+
+  Future<bool> identify(UserEntity user) async {
+    const uuid = Uuid();
+    final now = Timestamp.fromDateTime(DateTime.now());
+
+    final activities = [
+      api.Activity()
+        ..timestamp = now
+        ..id = uuid.v4()
+        ..type = api.ActivityType.ACTIVITY_IDENTIFY
+        ..userData = api.Activity_UserDataActivity(),
+    ];
+
+    final request = await createRequest()
+      ..activities.addAll(activities);
+
+    List<int> serializedRequest = request.writeToBuffer();
+    return await _apiCall(serializedRequest);
+  }
+
+  Future<bool> trackEvent(EventEntity event) async {
```

Comment:
Será que não valeria export uma interface para o usuário mandar N eventos para a dito ao mesmo tempo? Hoje caso ele tenha essa necessidade a gente força ele a mandar um por um, fazendo uma request de cada vez pro servidor, sendo que toda a modelagem da API é feita pra trabalhar com multiplas activities.

Reply:
não precisa, pois a ideia é que use as interfaces user, event e notification para interagir. O dev não precisa interagir diretamente com o canal da API.